elasticsearch/bulk/bulk.go

```diff
@@ -21,23 +25,27 @@ import (
 )
 
 type Bulk struct {
-	reader                 *bytes.Reader
 	errorLogger            logger.Logger
 	logger                 logger.Logger
+	metric                 *Metric
+	collectionIndexMapping map[string]string
+	batchKeys              map[string]int
 	dcpCheckpointCommit    func()
 	batchTicker            *time.Ticker
 	isClosed               chan bool
 	actionCh               chan document.ESActionDocument
 	esClient               *elasticsearch.Client
-	metric                 *Metric
-	collectionIndexMapping map[string]string
+	batch                  [][]byte
```

Comment:
You can also extract `[][]byte` into the custom type alias like

type BatchData [][]byte etc.

Reply:
It is not something that is reused. I don't want to make unnecessary imports.